x-pack/plugins/security_solution/public/detections/components/take_action_dropdown/index.tsx

```diff
@@ -166,6 +176,23 @@ export const TakeActionDropdown = React.memo(
       onInvestigateInTimelineAlertClick: closePopoverHandler,
     });
 
+    const osqueryAvailable = osquery?.isOsqueryAvailable({
+      agentId,
+    });
+
+    const handleOnOsqueryClick = useCallback(() => {
+      onOsqueryClick(agentId);
+      setIsPopoverOpen(false);
+    }, [onOsqueryClick, setIsPopoverOpen, agentId]);
+
+    const osqueryActionItem = useMemo(
+      () =>
+        OsqueryActionItem({
+          handleClick: handleOnOsqueryClick,
```

Comment:
That would be nice to test if this new action item is displayed (similar test to existing under the `should render take action items`) and maybe if onClick also handled properly.

Reply:
👍  thanks, will do!

Reply:
I added a check if item is displayed. If it comes to onClick - it's tested in Cypress. Would that be ok?